src/core/border_router/routing_manager.cpp

```diff
@@ -162,25 +164,53 @@ Error RoutingManager::GetNat64Prefix(Ip6::Prefix &aPrefix)
 }
 #endif
 
-Error RoutingManager::LoadOrGenerateRandomOmrPrefix(void)
+Error RoutingManager::LoadOrGenerateRandomBrUlaPrefix(void)
 {
     Error error = kErrorNone;
 
-    if (Get<Settings>().Read<Settings::OmrPrefix>(mLocalOmrPrefix) != kErrorNone || !IsValidOmrPrefix(mLocalOmrPrefix))
+    if (Get<Settings>().Read<Settings::BrUlaPrefix>(mBrUlaPrefix) != kErrorNone || !IsValidBrUlaPrefix(mBrUlaPrefix))
     {
-        Ip6::NetworkPrefix randomOmrPrefix;
+        Ip6::NetworkPrefix randomBrUlaPrefix;
 
-        LogNote("No valid OMR prefix found in settings, generating new one");
+        LogNote("No valid /48 BR ULA prefix found in settings, generating new one");
 
-        // TODO: generate OMR prefix from the /48 BR ULA prefix
-        error = randomOmrPrefix.GenerateRandomUla();
+        error = randomBrUlaPrefix.GenerateRandomUla();
         if (error != kErrorNone)
         {
-            LogCrit("Failed to generate random OMR prefix");
+            LogCrit("Failed to generate random /48 BR ULA prefix");
             ExitNow();
         }
 
-        mLocalOmrPrefix.Set(randomOmrPrefix);
+        randomBrUlaPrefix.m8[6] = 0;
+        randomBrUlaPrefix.m8[7] = 0;
+        mBrUlaPrefix.Set(randomBrUlaPrefix);
+        mBrUlaPrefix.SetLength(kBrUlaPrefixLength);
+
+        IgnoreError(Get<Settings>().Save<Settings::BrUlaPrefix>(mBrUlaPrefix));
+    }
+
+exit:
+    return error;
+}
+
+Error RoutingManager::LoadOrGenerateRandomOmrPrefix(void)
+{
+    Error error = kErrorNone;
+
+    if (Get<Settings>().Read<Settings::OmrPrefix>(mLocalOmrPrefix) != kErrorNone || !IsValidOmrPrefix(mLocalOmrPrefix))
+    {
+        LogNote("No valid OMR prefix found in settings, generating new one from /48 BR ULA prefix");
+
+        if (!IsValidBrUlaPrefix(mBrUlaPrefix))
+        {
+            SuccessOrExit(error = LoadOrGenerateRandomBrUlaPrefix());
+        }
```

Comment:
Considering that there are already in-field BRs which are already using our previous implementation, will the BR be changed to use the 48 bits prefix when the firmware is updated?

Reply:
Updated to rename `kKeyOmrPrefix` to `kKeyBrUlaPrefix` in the `Settings` enum. Thus the legacy saved prefix would invalid the `IsValidBrUlaPrefix` and get replaced by the new prefix.